test/model/src/driver/kv_workload_generator.cpp

```diff
@@ -471,6 +492,15 @@ kv_workload_generator::generate_transaction(size_t seq_no)
             probability_case(_spec.truncate)
             {
                 table_context_ptr table = choose_table(txn_ptr);
+
+                /*
+                 * Don't use truncate on FLCS tables, because a truncate on an FLCS table can
+                 * conflict with operations outside of the truncation range's key range. The
+                 * workload generator is not yet smart enough to account for that.
```

Comment:
I'm not sure how to explain it better. Basically, the problem is that let's say that your table has keys 10, 11, 12, 13, 14. Now let's say that you'd like to truncate 10—12 and update 13 at the same time in a different transaction. It turns out that with FLCS, this would result in `WT_ROLLBACK`, even though the two key ranges do not overlap. A concurrent update of 14 is not a problem because 14 is not adjacent to the truncation range.

This not a problem with VLCS or the row store. I did this fix in [WT-12539](https://jira.mongodb.org/browse/WT-12539), but I couldn't do it for FLCS because the fix depends on bounded cursors, which are not supported by FLCS. As a result, let's not test truncate on FLCS. I'm okay just leaving this as is—without a ticket to fix this behavior for FLCS or finding a way to update the test to not exercise that behavior—because MongoDB does not currently use FLCS. But it could be an option.

What would we a better way to write this?

Reply:
Maybe we can be more precise by following your example:
> Don't use truncate on FLCS tables, because a truncate on an FLCS table can conflict with operations outside of the truncation range's key range.

Should we mention here that the operation needs to be adjacent to the one of the ends of the truncate range?

Reply:
How about if we use the word "adjacent to" and then also include the above example, so that it's clear? I'm thinking something like this:
```
/*
 * Don't use truncate on FLCS tables, because a truncate on an FLCS table can
 * conflict with operations adjacent to the truncation range's key range. For
 * example, if a user wants to truncate range 10-12 on a table with keys [10, 11,
 * 12, 13, 14], a concurrent update to key 13 would result in a conflict (while an
 * update to 14 would be able proceed). This does not happen with the other table
 * types, which are implemented using bounded cursors; FLCS does not support bounded
 * cursors, so it uses a different implementation.
 *
 * The workload generator cannot currently account for this, so don't use truncate
 * with FLCS tables for now.
 */
```

Reply:
Looks good!

Reply:
Thanks for the explanation @pmacko86 ! Having a concrete example helps a lot.